clients/pubsub-common/src/test/java/org/eclipse/hono/client/pubsub/publisher/CachingPubSubPublisherFactoryTest.java

```diff
@@ -42,7 +44,8 @@ public class CachingPubSubPublisherFactoryTest {
     void setUp() {
         vertx = mock(Vertx.class);
         client = mock(PubSubPublisherClient.class);
-        factory = new CachingPubSubPublisherFactory(vertx, PROJECT_ID, null);
+        final FixedCredentialsProvider credentialsProvider = mock(FixedCredentialsProvider.class);
```

Comment:
I would rather have an interface being mocked here ...
That is also the reason why I was wondering if the PubSubSubscriber actually requires a *FixedCredentialsProvider* or also works with any *CredentialsProvider* (which I believe is the case) ....